### ISSUE TITLE
Unable to override header formatting

### ISSUE THREAD

**Issue by jbarreiros** [amzn/style-dictionary#612]
Hi, messing around with the latest 3.0-rc.8 release.

Creating a custom format that will output some HTML to be included in a project as a Twig file. In Twig, comments look like:
```twig
{# I'm a comment #}
```
The full header would be expected to look like:
```twig
{#
    Do not edit directly
    Generated on [date]
#}
```

The issue I'm noticing is that the `fileHeader()` function [accepts a `formatting` option](https://github.com/amzn/style-dictionary/blob/3.0/lib/common/formatHelpers/fileHeader.js#L47), but it appears to [ultimately be ignored](https://github.com/amzn/style-dictionary/blob/3.0/lib/common/formatHelpers/fileHeader.js#L69-L83).

For example:
```
// myCustomFormat.js
const header = fileHeader({ file, formatting: { prefix: '  ', header: '{#', footer: '#}' } });
```
The current logic appears to always overwrite the `prefix`, `header` and `footer` properties.

**Comment by jbarreiros** [comment]
I'm open to creating an MR, but wasn't sure my idea for fixing this is all that great. The only thing I can think of is:

```js
let {prefix, lineSeparator, header, footer} = Object.assign({}, defaultFormatting);

if (/* formattting object is not empty */) {
   let {prefix, lineSeparator, header, footer} = Object.assign({}, defaultFormatting, formatting);
   // The potentially annoying part here is that "header" and "footer" probably need to include
   // lineSeparator in their value. E.g. "footer: '\nEND\n'"
} else {
   // existing logic
}
```

**Comment by dbanksdesign** [comment]
Good catch. Sloppy logic on my part, and also lack of unit tests... Let me know your thoughts on #614. I just moved the default formatting options to the `defaultFormatting` object and removed the final `else`. Now it works as you would expect, but I might be missing a use case or something...

**Comment by dbanksdesign** [comment]
This is now fixed and released in 3.0.0-rc.9! I added a test with your use-case to make sure it worked.
